src/error.rs

```diff
@@ -1,21 +1,62 @@
 use std::num::ParseIntError;
+use std::fmt;
+use std::error;
 
 #[derive(Debug)]
 pub enum SdpParserError {
-    ParserLineError { message: String, line: String },
-    ParserUnsupported { message: String, line: String },
-    ParserSequence {
+    Line { message: String, line: String },
+    Unsupported { message: String, line: String },
+    Sequence {
         message: String,
         line: Option<usize>,
     },
+    Integer(ParseIntError),
 }
 
-impl From<ParseIntError> for SdpParserError {
-    fn from(_: ParseIntError) -> Self {
-        // TODO empty line error here makes no sense
-        SdpParserError::ParserLineError {
-            message: "failed to parse integer".to_string(),
-            line: "".to_string(),
+impl fmt::Display for SdpParserError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            SdpParserError::Line {
+                ref message,
+                ref line,
+            } => write!(f, "Line error: {} in line: {}", message, line),
+            SdpParserError::Unsupported {
+                ref message,
+                ref line,
+            } => write!(f, "Unsupported: {} in line: {}", message, line),
+            SdpParserError::Sequence { ref message, .. } => {
+                write!(f, "Sequence error: {}", message)
+            }
+            SdpParserError::Integer(ref err) => {
+                write!(f,
+                       "Integer parsing error: {}",
+                       error::Error::description(err))
+            }
+        }
+    }
+}
+
+impl error::Error for SdpParserError {
+    fn description(&self) -> &str {
+        match *self {
+            SdpParserError::Line { ref message, .. } |
+            SdpParserError::Unsupported { ref message, .. } |
+            SdpParserError::Sequence { ref message, .. } => message,
+            SdpParserError::Integer(ref err) => error::Error::description(err),
+        }
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
+        match *self {
+            SdpParserError::Integer(ref err) => Some(err),
+            // TODO is None really the right thing for our own errors?
```

Comment:
It seems appropriate. I can't think of what else cause would return. You wouldn't want cause to return `&self` in my opinion, because that would just make life difficult for someone trying to output or record a chain of errors.